src/hdf5_back_gen.py

```diff
@@ -23,12 +39,12 @@
 INDENT = '    '
 
 def convert_canonical(raw_list):
-    if isinstance(raw_list, str):
+    if type(raw_list) is str:
```

Comment:
I think that is instance is better in general.  Or was that causing the problem?  unicode is not a subtype of str, so the original line is fine.